master/internal/resourcemanagers/agent/agents.go

```diff
@@ -70,6 +77,18 @@ func (a *agents) Receive(ctx *actor.Context) error {
 			}
 		}
 	case api.WebSocketConnected:
+		cmuxConn := connsave.GetConn(msg.Ctx.Request().Context()).(*cmux.MuxConn)
+		// Here, we just have to check that there are any certificates at all, since the top-level TLS
+		// config verifies that any certificates that are provided are valid.
+		if tlsConn, ok := cmuxConn.Conn.(*tls.Conn); ok {
+			if a.config.RequireAuthentication && len(tlsConn.ConnectionState().PeerCertificates) == 0 {
```

Comment:
btw we have a [singleton](https://github.com/determined-ai/determined/blob/master/master/internal/config/config.go#L140) for master config
so instead of passing config around into all the objects that need it, you can just use `config.GetMasterConfig().AgentRM.RequireAuthentication` in place